File: client/src/app/modules/miscellaneous/services/department.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Department } from '../../miscellaneous/model/Department';

@Injectable({
  providedIn: 'root'
})
export class DepartmentService extends AppRestService<Department>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getDepartments() {
    return this.getListCached(new HttpParams(), "departments");
  }

  addOrUpdateDepartment(department: Department) {
    this.clearListCache(new HttpParams(), "departments");
    return this.addOrUpdate(new HttpParams(), "department", department, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/DepartmentService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.Department;

public interface DepartmentService {
    public List<Department> getDepartments();

    public Department getDepartment(Integer id);

    public Department addOrUpdateDepartment(Department department);
}
